### DISCUSSION TITLE
PlayerBot

### DISCUSSION THREAD

**Discussion by creekers** [azerothcore/azerothcore-wotlk#12561] [Modules]
Hi :)
Sorry, i'm french user ;) i have a probleme when try to compil the source code with playerbot module.
Vs 2019 16.11 return this lot of error :

22>PlayerbotCommandServer.cpp
22>Please define _WIN32_WINNT or _WIN32_WINDOWS appropriately. For example:
22>- add -D_WIN32_WINNT=0x0601 to the compiler command line; or
22>- add _WIN32_WINNT=0x0601 to your project's Preprocessor Definitions.
22>Assuming _WIN32_WINNT=0x0601 (i.e. Windows 7 target).
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\strategy\actions\MovementActions.cpp(1051,37): error C2039: 'getThreatMgr' is not a member of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\src\server\game\Entities\Vehicle\Vehicle.h(27): message : see the statement of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\strategy\values\AttackerCountValues.cpp(34,19): error C2039: 'getThreatMgr' is not a member of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\src\server\game\Spells\Spell.h(27): message : see the statement of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\strategy\values\AttackersValue.cpp(75,24): error C2039: 'getThreatMgr' is not a member of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\ServerFacade.h(11): message : see the statement of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\strategy\values\AttackersValue.cpp(75,68): error C2039: 'getThreatMgr' is not a member of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\ServerFacade.h(11): message : see the statement of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\strategy\values\AttackersValue.cpp(75,111): error C2039: 'getThreatMgr' is not a member of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\ServerFacade.h(11): message : see the statement of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\strategy\values\AttackersValue.cpp(110,37): error C2039: 'getThreatMgr' is not a member of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\ServerFacade.h(11): message : see the statement of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\strategy\values\AttackersValue.cpp(134,59): error C2039: 'getThreatMgr' is not a member of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\ServerFacade.h(11): message : see the statement of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\strategy\values\AttackersValue.cpp(149,44): error C2039: 'getThreatMgr' is not a member of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\ServerFacade.h(11): message : see the statement of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\strategy\values\TargetValue.cpp(24,38): error C2039: 'getThreatMgr' is not a member of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\src\server\game\Spells\Spell.h(27): message : see the statement of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\strategy\values\TargetValue.cpp(24,30): error C2530: 'ThreatMgr' : references must be initialized
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\strategy\values\ThreatValues.cpp(45,31): error C2039: 'getThreatMgr' is not a member of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\src\server\game\Spells\Spell.h(27): message : see the statement of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\modules\mod-playerbots\src\strategy\values\ThreatValues.cpp(58,36): error C2039: 'getThreatMgr' is not a member of 'Unit'
22>F:\Wow_Serveur\Sources\AzerothCore-PlayerBots\src\server\game\Spells\Spell.h(27): message : see the statement of 'Unit'

**Comment by ZhengPeiRu21** [comment]
The Playerbot module must be built from a special fork of AzerothCore because it is use some functionality that is not yet added through hooks. Please ensure to be building it with the correct fork: https://github.com/ZhengPeiRu21/azerothcore-wotlk/tree/Playerbot

**Reply by creekers** [reply]
Hi, i am using your playerbot branche, its a fresh compil after your last commit ;)

**Reply by ZhengPeiRu21** [reply]
Please ensure both the module and the branch are up to date. It looks like you have outdated code. getThreatMgr was changed to GetThreatMgr.

**Reply by creekers** [reply]
thanks for your help ;) probleme solved

**Comment by creekers** [comment]
Hi, 
Everything works thanks for the compilation info ;) I compiled the playerbot branch with the solocraft and lfg modules, everything works except two problems: in solo instance the mobs are untouchable and dodge attacks, and second problem, we can't join an instance with group bots or not. it's normal ?
thanks ;)
